<?xml version="1.0"?>
<launch>
    <arg name="limited" default="false"  doc="If true, limits joint range [-PI, PI] on all joints." />
    <arg name="paused" default="false" doc="Starts gazebo in paused mode" />
    <arg name="gui" default="true" doc="Starts gazebo gui" />
    <arg name="sim" default="true" />
    <arg name="debug" default="false"/>
 
    <!-- Remap follow_joint_trajectory -->
    <remap if="$(arg sim)" from="/follow_joint_trajectory" to="/arm_controller/follow_joint_trajectory"/>
    <remap unless="$(arg sim)" from="/follow_joint_trajectory" to="/scaled_pos_joint_traj_controller/follow_joint_trajectory"/>

    <!-- Run the main MoveIt executable without trajectory execution (we do not have controllers configured by default) -->
    <include file="$(find ur10_moveit_config)/launch/move_group.launch">
        <arg name="allow_trajectory_execution" value="true"/>
        <arg name="fake_execution" value="false"/>
        <arg name="info" value="true"/>
        <arg name="debug" value="$(arg debug)"/>
    </include>

    <!-- joystick node -->
    <!-- <node name="joy_command" pkg="ur10_python_interface" type="joystick.py" output="log"/> -->

    <param name="interface" type="str" value="not ready"/>

    <!-- ur10 python interface node -->
    <node if="$(arg sim)" name="robot_interface" pkg="ur10_python_interface" type="move_group_python_interface.py" args="sim" output="screen"/>    
    <node unless="$(arg sim)" name="robot_interface" pkg="ur10_python_interface" type="move_group_python_interface.py" args="real" output="screen"/>    

    <!-- mode manager node -->
    <node name="mode_manager" pkg="ur10_python_interface" type="mode_manager.py" output="log"/>
    

</launch>
